src/app/api/expenses/route.ts

```diff
@@ -38,17 +38,13 @@ export async function POST(request) {
         const docSnap = await getDoc(docRef);
 
         if(docSnap.exists()) { 
-            const expenses = docSnap.data().expenses;
-            const expenseIndex = expenses.findIndex((exp) => exp.id === expense.id);
-            expenses[expenseIndex] = {...expense};
-    
             await updateDoc(docRef, {
-                expenses,
+                expenses: arrayUnion(expense)
```

Comment:
Arrayunion was used but was not imported at top of file causing expenses to not get added.